xmppserver/src/main/java/org/jivesoftware/openfire/websocket/XmppWebSocket.java

```diff
@@ -162,6 +162,8 @@ void closeSession() {
             closeStream(null);
         }
         if (xmppSession != null) {
+            xmppSession.getStreamManager().formalClose();
+
```

Comment:
This does not seem right. Effectively, we've allowed resumption to be negotiated, but when the session is closed, we disregard that completely, and instead simply close the session.

It would be a lot better to prevent resumption from being negotiated for these kind of sessions in the first place. I think that can be done by modifying `org.jivesoftware.openfire.streammanagement.StreamManager#allowResume`

Reply:
> It would be a lot better to prevent resumption from being negotiated for these kind of sessions in the first place. I think that can be done by modifying `org.jivesoftware.openfire.streammanagement.StreamManager#allowResume`

Isn't that a global setting for BOSH as well as TLS over 5222?

Reply:
Thanks for the quick videoconf to clarify.I will modify allowResume to return false for websockets and then use it in XmppWebSocket#closeSession  with getResume to determine if the xmpp session should be closed or not